sdk/agent/agent.go

```diff
@@ -90,6 +100,32 @@ func NewAgent(c Config) *Agent {
 	return agent
 }
 
+// Snapshot is a snapshot of the agent and its dependencies excluding any fields
+// provided in the Config when instantiating an agent. A Snapshot can be
+// restored into an Agent using NewAgentWithSnapshot.
+type Snapshot struct {
+	OtherEscrowAccount       *keypair.FromAddress
+	OtherEscrowAccountSigner *keypair.FromAddress
+	StreamerCursor           string
+	State                    *struct {
+		Initiator bool
```

Comment:
❓ is there a reason for putting `Inititiator` on the agent Snapshot? Instead of the Channel Snapshot since it's defined on state.Channel?

Reply:
The reason is that the `Initiator` is not internal state of `state.Channel`, it's external config that the caller provides when setting it up. The `state.Snapshot` only stores internal state, not initialization config.